infrastructure/repository/project_impl.go

```diff
@@ -216,25 +217,52 @@ func (r *ProjectRepository) AddProjectMembers(ctx context.Context, projectID uui
 		return repository.ErrInvalidArg
 	}
 
-	// ユーザーの重複チェック
-	projectMembersMap := make(map[uuid.UUID]struct{}, len(projectMembers))
-	for _, v := range projectMembers {
-		if _, ok := projectMembersMap[v.UserID]; ok {
-			return repository.ErrInvalidArg
-		}
-		projectMembersMap[v.UserID] = struct{}{}
-	}
+	projectRaw := new(model.Project)
 
 	// プロジェクトの存在チェック
 	err := r.h.
 		WithContext(ctx).
 		Where(&model.Project{ID: projectID}).
-		First(&model.Project{}).
+		First(&projectRaw).
 		Error
 	if err != nil {
 		return err
 	}
 
+	// 追加するユーザーの期間がプロジェクトの期間内かどうか
+	project := &domain.ProjectDetail{
+		Project: domain.Project{
+			ID:   projectID,
+			Name: projectRaw.Name,
+			Duration: domain.YearWithSemesterDuration{
+				Since: domain.YearWithSemester{
+					Year:     int(projectRaw.SinceYear),
+					Semester: int(projectRaw.SinceSemester),
+				},
+				Until: domain.YearWithSemester{
+					Year:     int(projectRaw.UntilYear),
+					Semester: int(projectRaw.UntilSemester),
+				},
+			},
+		},
+		Description: projectRaw.Description,
+		Link:        projectRaw.Link,
+	}
+
+	// 追加するユーザー群について
+	projectMembersMap := make(map[uuid.UUID]struct{}, len(projectMembers))
+	for _, v := range projectMembers {
+		if !project.Duration.Includes( // プロジェクトの期間内かどうか
+			domain.NewYearWithSemesterDuration(
+				v.SinceYear, v.SinceSemester, v.UntilYear, v.UntilSemester)) {
+			return repository.ErrInvalidArg
+		}
```

Comment:
個人的にあんまりこのフォーマットが好きじゃない
↑のレビューと合わせると↓みたいに書きたいかな
```suggestion
		memberDuration := domain.NewYearWithSemesterDuration(v.SinceYear, v.SinceSemester, v.UntilYear, v.UntilSemester) // NOTE: ここは長いから引数ごとに改行してもいいかも
		// プロジェクトの期間内かどうか
		if !projectDuration.Includes(memberDuration) {
			return repository.ErrInvalidArg
		}
```